vm/actor/src/builtin/miner/mod.rs

```diff
@@ -415,16 +613,38 @@ impl Actor {
         BS: BlockStore,
         RT: Runtime<BS>,
     {
-        if params.expiration <= rt.curr_epoch() {
+        if !check_supported_proof_types(params.seal_proof) {
+            return Err(ActorError::new(
+                ExitCode::ErrIllegalArgument,
+                format!("unsupported seal proof type: {:?}", params.seal_proof),
+            ));
+        }
+
+        #[allow(clippy::absurd_extreme_comparisons)]
+        if params.sector_number > MAX_SECTOR_NUMBER {
```

Comment:
i know they do this in lotus, but can this ACTUALLY ever occur in Rust?

Reply:
It cannot, hence the clippy thing, but this seems like the safest way to do it in case they lower that constant later?

Reply:
Makes sense. Its just bizarre haha

Reply:
The bound is based on a limitation of their Amt implementation. This definitely needs to be checked because otherwise a param could just be passed in to create a discrepancy.

Thanks for flagging this, I noticed that they have changed the max in the Amt implementation. I'll fix that now